geode-lucene/src/test/java/org/apache/geode/cache/lucene/internal/cli/LuceneIndexCommandsWithReindexAllowedIntegrationTest.java

```diff
@@ -63,21 +63,22 @@ public void whenLuceneReindexingInProgressThenListIndexCommandMustExecuteSuccess
     csb.addOption(LuceneCliStrings.LUCENE__REGION_PATH, REGION_NAME);
     csb.addOption(LuceneCliStrings.LUCENE_CREATE_INDEX__FIELD, "__REGION_VALUE_FIELD");
 
-    serverVM.invoke(() -> createRegion());
-    AsyncInvocation ai = serverVM.invokeAsync(() -> {
+    createRegion();
+    Thread ai = new Thread(() -> {
       int count = 0;
       while (true) {
-        getCache().getRegion(REGION_NAME).put(count, "hello world" + count);
+        server.getCache().getRegion(REGION_NAME).put(count, "hello world" + count);
       }
     });
 
+    ai.start();
+
     gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess()
         .containsOutput("Successfully created lucene index");
     csb = new CommandStringBuilder(LuceneCliStrings.LUCENE_LIST_INDEX);
     csb.addOption(LuceneCliStrings.LUCENE_LIST_INDEX__STATS, "true");
     gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess();
-    ai.cancel(true);
-
+    ai.interrupt();
```

Comment:
I don't think this is the right way to stop this thread. It may never actually see the interrupt unless something checks, or it may interrupt the thread doing IO and cause failures in later tests.

Instead of this, cancel the task with an AtomicBoolean or something like that.

Reply:
fixed